### ISSUE TITLE
TH2F is not read properly from file (does not agree with pyroot)

### ISSUE THREAD

**Issue by lgray** [scikit-hep/uproot4#234]
The bin edge meanings are not transposed, but rather the values are changed!
Uproot4 indicates the histogram is a `TH2F (version 3)`

```python3
import numpy as np
import uproot
import uproot3  # NB: uproot3 agrees with pyroot
f = uproot.open('https://github.com/CoffeaTeam/coffea/raw/master/tests/samples/testSF2d.histo.root')
hist = f['scalefactors_Tight_Electron']
edges = hist.axes[0].edges()

print(edges)

f3 = uproot3.open('https://github.com/CoffeaTeam/coffea/raw/master/tests/samples/testSF2d.histo.root')
hist3 = f[b'scalefactors_Tight_Electron']
edges3 = hist3.edges[0]

print(edges3)

assert np.all(edges == edges3)
```
gives you:
```
[-2.5 -2.  -1.5 -1.  -0.5  0.   0.5  1.   1.5  2.   2.5]
[-2.5   -2.    -1.566 -1.444 -0.8    0.     0.8    1.444  1.566  2.
  2.5  ]
Traceback (most recent call last):
  File "bug.py", line 17, in <module>
    assert np.all(edges == edges3)
AssertionError
```

**Comment by lgray** [comment]
ah, this is fixed in uproot 4.0.1rc2.

**Comment by jpivarski** [comment]
I couldn't reproduce this; @lgray updated and the problem apparently went away.

For reference,

```python
>>> import uproot
>>> import uproot3
>>> uproot.__version__
'4.0.1rc2'
>>> uproot3.__version__
'3.14.1'

>>> hist_uproot4 = uproot.open("https://github.com/CoffeaTeam/coffea/raw/"
                               "master/tests/samples/testSF2d.histo.root")["scalefactors_Tight_Electron"]
>>> hist_uproot4.axes[0].edges()
array([-2.5  , -2.   , -1.566, -1.444, -0.8  ,  0.   ,  0.8  ,  1.444,
        1.566,  2.   ,  2.5  ])
>>> hist_uproot4.axes[1].edges()
array([ 10.,  20.,  35.,  50.,  90., 150., 500.])

>>> hist_uproot3 = uproot3.open("https://github.com/CoffeaTeam/coffea/raw/"
                               "master/tests/samples/testSF2d.histo.root")["scalefactors_Tight_Electron"]
>>> hist_uproot3.edges[0]
array([-2.5  , -2.   , -1.566, -1.444, -0.8  ,  0.   ,  0.8  ,  1.444,
        1.566,  2.   ,  2.5  ])
>>> hist_uproot3.edges[1]
array([ 10.,  20.,  35.,  50.,  90., 150., 500.])
```
